### PR TITLE
Add ability to use heuristic from given solution.

### PR DESCRIPTION
**Pull request by drvinceknight**
This allows for a given solution to be further optimised using a
heuristic approach.

### CODE REVIEW

**Comment by drvinceknight** [comment]
From travis:

```
0.36s$ coverage report -m
Name                                                         Stmts   Miss  Cover   Missing
------------------------------------------------------------------------------------------
src/conference_scheduler/__init__.py                             0      0   100%
src/conference_scheduler/converter.py                           17      0   100%
src/conference_scheduler/heuristics/__init__.py                  3      0   100%
src/conference_scheduler/heuristics/hill_climber.py             18      0   100%
src/conference_scheduler/heuristics/simulated_annealing.py      27      0   100%
src/conference_scheduler/heuristics/utils.py                    26      0   100%
src/conference_scheduler/lp_problem/__init__.py                  2      0   100%
src/conference_scheduler/lp_problem/constraints.py              45      0   100%
src/conference_scheduler/lp_problem/objective_functions.py      18      0   100%
src/conference_scheduler/lp_problem/utils.py                    64      0   100%
src/conference_scheduler/resources.py                           64      0   100%
src/conference_scheduler/scheduler.py                           60      0   100%
src/conference_scheduler/validator.py                           25      0   100%
------------------------------------------------------------------------------------------
TOTAL                                                          369      0   100%
```

File: src/conference_scheduler/scheduler.py [hunk 1]
```diff
@@ -74,18 +76,23 @@ def heuristic(events,
...
+        X = algorithm(initial_array=X,
+                      objective_function=count_violations,
+                      lower_bound=0,
+                      **initial_solution_algorithm_kwargs)
+    else:
+        X = initial_solution
 
     if objective_function is not None:
 
+        if objective_function is of.equity_capacity_demand_difference:
```
Comment (drvinceknight) [inline src/conference_scheduler/scheduler.py]: I can do without this check @meatballs, so just have that `"beta"` is passed to all objective functions whether or not it is needed. I think that might be nicer: any preference?

Comment (meatballs) [reply src/conference_scheduler/scheduler.py]: I reckon without is nicer

Comment (meatballs) [reply src/conference_scheduler/scheduler.py]: sorry, without the condition - pass the beta whether it's needed or not

Comment (drvinceknight) [reply src/conference_scheduler/scheduler.py]: Yup! Doing that now.

**Pull request merged**